lib/private/shell.ts

```diff
@@ -1,11 +1,33 @@
 import * as child_process from 'child_process';
+import { EventType } from '../progress';
+import { SubprocessOutputDestination } from './asset-handler';
 
-export type Logger = (x: string) => void;
+export type ShellEventType = 'open' | 'data_stdout' | 'data_stderr' | 'close';
+
+export type ShellEventPublisher = (event: ShellEventType, message: string) => void;
 
 export interface ShellOptions extends child_process.SpawnOptions {
-  readonly quiet?: boolean;
-  readonly logger?: Logger;
+  readonly shellEventPublisher: ShellEventPublisher;
   readonly input?: string;
+  readonly subprocessOutputDestination?: SubprocessOutputDestination;
+}
+
+/**
+ * A helper function to convert shell events to asset progress events
+ * @param event a shell event
+ * @returns an {@link EventType}
+ */
+export function shellEventToEventType(event: ShellEventType): EventType {
```

Comment:
Missed this on the first go-round, but this is putting knowledge of asset publishing and asset progress reporting into `shell.ts`. 

In [this comment](https://github.com/cdklabs/cdk-assets/pull/221#discussion_r1908397901) I said:

> The "dependencies" strictly go to the right:
>
> * CLI knows about, and can depend on, asset publishing (but not the reverse!)
> * Asset publishing knows about, and can depend on, the shell helper (but not the reverse!)

By putting this function here, the shell helper knows about asset publishing, and the dependency arrow points the wrong way. The fix is simple, but this file in the asset publishing code.

Reply:
Yeah moved this to `progress.ts` as that's where we describe the EventType as well